mrp_mto_owner/models/stock_rule.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _prepare_mo_vals(
+        self,
+        product_id,
+        product_qty,
+        product_uom,
+        location_dest_id,
+        name,
+        origin,
+        company_id,
+        values,
+        bom,
+    ):
+        vals = super(StockRule, self)._prepare_mo_vals(
+            product_id,
+            product_qty,
+            product_uom,
+            location_dest_id,
+            name,
+            origin,
+            company_id,
+            values,
+            bom,
+        )
+        move_dest_ids = values.get("move_dest_ids")
+        if move_dest_ids:
+            origin_move = self._find_origin_move(move_dest_ids[0])
+            if origin_move and origin_move.raw_material_production_id.owner_id:
+                vals["owner_id"] = origin_move.raw_material_production_id.owner_id.id
+        return vals
```

Comment:
@AungKoKoLin1997 Do you think we can simply do this, discarding the _find_origin_move() method?  I guess we don't need to get to the very origin, as long as the parent inherits the value from its parent. Same for the `purchase_mto_owner` module.

```suggestion
        if move_dest_ids:
            vals["owner_id"] = move_dest_ids[0].raw_material_production_id.owner_id.id
        return vals
```

Reply:
Yes. I will check and if the parent inherits the value from its parent, I will get rid of `_find_origin_move`.